BEPs/BEP322.md

```diff
@@ -0,0 +1,487 @@
+
+- **Improve Stability**: Unifying specifications for both validators and MEV providers enhances integration and
+  stabilizes the market.
+
+- **Improve Economy**: A unified architectural specification increases speculators' costs, fostering healthy market
+  competition.
+  Validators can extract more value by integrating with multiple MEV providers, benefiting delegators as well.
+
+- **Improve Transparency**: This specification aims to bring transparency to the BSC MEV market, exposing profit
+  distribution among stakeholders to the public.
+
+## 3. Background
+
+Before delving into the specification, let's analyze the BSC trust model and its distinguishing consensus protocol.
+
+### 3.1 BSC Trust Model
+
+Validators in BNB Smart Chain are more trustworthy, requiring substantial BNB delegation and maintaining a high
+reputation.
+This is in contrast to Ethereum, where becoming a validator is relatively easier.
+In BSC, there are about 40 validators (more validators are coming) with 20M BNB staked, including active and inactive
+ones. In Ethereum, the barrier to becoming a validator is very low (i.e., 32 ETH), and there are more
+than [800 thousand Ethereum validators](https://www.nasdaq.com/articles/the-most-pressing-issue-on-ethereum-is-validator-size-growth).
+
+Meanwhile, the relay role in Ethereum Builder API is supposed to be trusted by both builders and validators.
+In BSC, misbehaviour of validators will lead to reputation damage and un-delegations, the necessity of introducing
+another trusted role is not that much. In this specification, we don't utilize relay; however, it is still workable to
+dd relay between builders and validators.
+
+### 3.2 BSC Consensus
+
+In Ethereum, a block header is passed from a builder to a validator for sign, and then a block can be broadcast to the
+network
+without revealing the transactions to a validator. Unlike Ethereum, generating a valid block header in BSC involves
+executing transactions and system contract calls (e.g., transfer coinbase, deposit to the validator set contract),
+making it impossible for builders to broadcast block.
+
+## 4 Workflow
+
+### 4.1 Definition
+
+- **Builder**: party specialized in the construction of ethereum execution payloads using transactions received from
+  users and searchers (trusted by searchers and users for fair inclusion)
+
+- **Proposer**: party which signs and submits a beacon block to the network, often referred to as the validator.
+
+- **Mev-sentry**: party who runs in front of a validator to enhance network security and proxy requests.
+
+- **Bid**: used by a builder to do competitive bidding for a block space. A bid should include the following
+  information:
+    - **parent_hash**. The hash of the parent block.
+    - **gas_value**. The gas value of the block.
+    - **builder_fee_value**. The fee that builder would like to take.
+    - **builder_address**. The address of the builder.
+
+### 4.2 One-round interaction (default)
```

Comment:
One-round interaction with sentry node can also supported - we can make this clear as two-round wit sentry node.
In real adoption, one round with sentry node could be the default.